File: bq-metrics-extractor/bq-metrics-extractor-server/src/main/java/org/ostelco/bqmetrics/BqMetricsExtractorApplication.java
package org.ostelco.bqmetrics;

import io.dropwizard.Application;
import io.dropwizard.setup.Bootstrap;
import io.dropwizard.setup.Environment;
import org.ostelco.bqmetrics.backend.DatabaseBackend;

/**
 * Main entry point to the bq-metrics-extractor API server.
 */
public final class BqMetricsExtractorApplication extends Application<BqMetricsExtractorConfiguration> {

    public static void main(final String[] args) throws Exception {
        new BqMetricsExtractorApplication().run(args);
    }

    @Override
    public void initialize(Bootstrap<BqMetricsExtractorConfiguration> bootstrap) {

    }

    @Override
    public void run(final BqMetricsExtractorConfiguration configuration, final Environment environment) {
        //DatabaseBackend databaseBackend = configuration.getDatabaseBackend(environment);
    }
}

File: bq-metrics-extractor/bq-metrics-extractor-server/src/main/java/org/ostelco/bqmetrics/BqMetricsExtractorApplication.kt
package org.ostelco.bqmetrics

import io.dropwizard.Application
import io.dropwizard.setup.Bootstrap
import io.dropwizard.setup.Environment
import org.ostelco.bqmetrics.backend.DatabaseBackend

/**
 * Main entry point to the bq-metrics-extractor API server.
 */
class BqMetricsExtractorApplication : Application<BqMetricsExtractorConfiguration>() {

    override fun initialize(bootstrap: Bootstrap<BqMetricsExtractorConfiguration>?) {

    }

    override fun run(configuration: BqMetricsExtractorConfiguration, environment: Environment) {
        //DatabaseBackend databaseBackend = configuration.getDatabaseBackend(environment);
    }

    companion object {

        @Throws(Exception::class)
        @JvmStatic
        fun main(args: Array<String>) {
            BqMetricsExtractorApplication().run(*args)
        }
    }
}
